File: project-euler/js/008.js
const { parse } = require('path');
const readline = require('readline')

const rl = readline.createInterface({input: process.stdin, output: process.stdout});
rl.prompt();

let input = '' 

rl.on('line', (inputLine) => {
    input+=(inputLine+'\n');
})

rl.on('close', () => {
    main(input);
    process.exit(0);
})

function largestProdSeries(numDigits, numDigitsProd, n) {
    let largestProd = 0
    const array = n.toString().split('').map(elem => parseInt(elem))
    for (let i = 0; i < numDigits - numDigitsProd + 1; i++) {
        let tempProd = 1
        for (let j = 0; j < numDigitsProd; j++) {
            tempProd *= array[i+j]            
        }
        largestProd = Math.max(largestProd, tempProd)
    }
    return largestProd
}

function main(input) {

    const inputArray = input
        .split('\n')
        .filter(elem => elem !== '')
        .map(elem => elem.split(' ')
            .map(elem => parseInt(elem)))

    const numInputs = inputArray[0][0];

    for(var i = 1; i < numInputs+1; i+=2) {
        console.log(largestProdSeries(inputArray[i][0], inputArray[i][1], inputArray[i+1][0]))
    }

}

File: project-euler/python/008.py
#!/bin/python3
import sys
from functools import reduce

def largest_prod_series(n_digits, n_digits_prod, n):
    largest_prod = 0
    arr = list(map(int, list(str(n))))
    for i in range(n_digits - n_digits_prod + 1):
        sub_arr = arr[i:i+n_digits_prod]
        temp_prod = reduce((lambda x, y: x*y), sub_arr)
        largest_prod = max(largest_prod, temp_prod)
    return largest_prod

num_inputs = int(input().strip())
for _ in range(num_inputs):
    n_digits, n_digits_prod = map(int, input().strip().split(' '))
    n = input().strip()
    print(largest_prod_series(n_digits, n_digits_prod, n))